text/0894-eslint-plugin-import.md

```diff
@@ -0,0 +1,174 @@
+* [eslint](https://eslint.org/) - General JavaScript best practices
+* [eslint-plugin-ember](https://github.com/ember-cli/eslint-plugin-ember) - Ember best practices
+* [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node) - Node best practices
+* [eslint-plugin-qunit](https://github.com/platinumazure/eslint-plugin-qunit) - Testing best practices ([RFC](https://github.com/emberjs/rfcs/blob/master/text/0702-eslint-plugin-qunit.md))
+* [prettier](https://prettier.io/) - Code styling ([RFC](https://github.com/emberjs/rfcs/blob/master/text/0628-prettier.md))
+* [stylelint](https://stylelint.io/) - CSS styling ([RFC](https://github.com/emberjs/rfcs/blob/master/text/0814-stylelint.md))
+
+But there's an important aspect of Ember apps that is not yet targeted by specialized linting: import statements. Imports are present in almost every file, and various problems or even crashes can result from invalid imports.
+
+[eslint-plugin-import](https://github.com/import-js/eslint-plugin-import) is one of the top few, most popular ESLint plugins, with around 8 million dependents on GitHub, 4.5k stars on GitHub, and 17 million weekly downloads on NPM, as of this writing. It contains 40+ rules for detecting problems with imports as well as enforcing usage preferences.
+
+## Detailed design
+
+The general idea is that we will update the `app` and `addon` blueprints to add the eslint-plugin-import package to the `package.json`, and update the linting configuration to extend the relevant configurations.
+
+### Packages
+
+The following packages will be added to the `package.json` of both `app` and `addon` blueprints:
+
+* [eslint-plugin-import](https://www.npmjs.com/package/eslint-plugin-import)
+
+### Configuration changes
+
+The ESLint config file that is generated will be updated to extend the `recommended` linting configuration:
+
+```js
+// .eslintrc.js
+{
+  extends: ['plugin:import/recommended'],
+}
+```
+
+We can also configure the [ember-cli-typescript](https://github.com/typed-ember/ember-cli-typescript) blueprint to enable the `typescript` config, but only for TypeScript files:
+
+```js
+// .eslintrc.js
+{
+  overrides: [
+    {
+      files: ['*.ts'],
+      extends: ['plugin:import/typescript'],
+    },
+  ],
+}
+```
+
+Note that these configurations enable only a handful of key rules from the plugin's 40+ rules. We may optionally choose to individually enable additional rules that we determine to be particularly valuable for Ember apps, either in this RFC, or in the future. But regardless, getting the plugin with its basic linting setup into place right now is worthwhile on its own, as this will also make it much easier for users to enable rules on their own too.
+
+### Resolver
+
+There are a few issues to fix with the [import/no-unresolved](https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/no-unresolved.md) rule.
```

Comment:
Did some testing on this plugin today in my own https://github.com/NullVoxPopuli/eslint-configs

`import/no-unresolved` does not understand `package.json#exports` -- which is a bit of a deal breaker for modern packages.

I tested with `vitest`:

```
eslint-configs/vitest.config.ts
  1:30  error  Unable to resolve path to module 'vitest/config'  import/no-unresolved
# ...
❯ cat node_modules/vitest/package.json | jq '.exports';
{
  ... ✂️...
  "./config": {
    "types": "./config.d.ts",
    "require": "./dist/config.cjs",
    "import": "./dist/config.js"
  }
  
```

So far, all V2 addons also use `package.json#exports`

Reply:
Thanks for testing that. I have updated the RFC to disable the `import/no-unresolved` lint rule.